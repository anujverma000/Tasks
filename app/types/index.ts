export interface TodoCategoryProps {
  title: string;
  description: string;
  taskCount: number;
  id: string;
}

export interface TodoProps {
  id: string;
  title?: string;
  description?: string;
  completed: boolean;
  category?: TodoCategoryProps;
}

export type NotesProps = {
  title: string;
  description: string;
};
