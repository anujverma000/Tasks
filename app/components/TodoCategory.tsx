import React from 'react';
import {StyleSheet, Text} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import {TodoCategoryProps} from '../types';

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#FF6A6F',
    padding: 16,
    marginVertical: 12,
    marginHorizontal: 4,
    width: 120,
    borderRadius: 6,
    elevation: 1,
  },
  selectedContainer: {
    paddingBottom: 42,
  },
  title: {
    fontSize: 14,
    color: '#8C8080',
  },
  selectedTitle: {
    fontSize: 16,
    color: 'white',
    fontWeight: 'bold',
  },
});

export interface TodoCategoryDisplyProps extends TodoCategoryProps {
  selected: boolean;
}
const TodoCategory = ({title, selected}: TodoCategoryDisplyProps) => {
  const gradientColors = selected
    ? ['#fe9168', '#ff4a73']
    : ['#ffffff', '#f2f2f2'];

  return (
    <LinearGradient
      colors={gradientColors}
      style={[styles.container, selected ? styles.selectedContainer : {}]}
      start={{x: 0, y: 0}}
      end={{x: 0, y: 1}}>
      <Text style={[styles.title, selected ? styles.selectedTitle : {}]}>
        {title}
      </Text>
    </LinearGradient>
  );
};

export default TodoCategory;
