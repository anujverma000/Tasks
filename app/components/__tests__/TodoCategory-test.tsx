import React from 'react';
import renderer from 'react-test-renderer';
import {TodoCategoryProps} from '../../types';
import TodoCategory, {TodoCategoryDisplyProps} from '../TodoCategory';

it('TodoCategory renders correctly', () => {
  const category: TodoCategoryProps = {
    title: 'category Title',
    description: 'category description',
    id: 'category-id',
  };
  const props: TodoCategoryDisplyProps = {
    ...category,
    selected: true,
  };
  const tree = renderer.create(<TodoCategory {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
