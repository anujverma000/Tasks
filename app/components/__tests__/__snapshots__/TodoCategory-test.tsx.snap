// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoCategory renders correctly 1`] = `null`;
